### DISCUSSION TITLE
how to implement satellite.at(time).is_sunlit(TLE)?

### DISCUSSION THREAD

**Discussion by dododo42** [skyfielders/python-skyfield#744] [Q&A] [answered]
First of all, I'd like to thank Brandon Rhodes for the awsome library.
I have just started using Python and it has already been so usefull.

Now, regarding the question:
I have the TLEs of a satellite which has mounted a sensor on it.
I need to retrieve some data about this sensor, such as if it can see the Sun or not.
I was planning to do this by myself, before noticing the existence of is_sunlit(ephemeris).
Unfortunately, the ephemeris of the satellite are not available, and I am working with TLEs only.
Is there a way to use is_sunlit() using the TLE as an input?
I tried to search for this through the documentation but I did not find anything.

At this point, my approach would be:
1. retrieve satellite position in ECI coordinates from TLE at time t
2. retrieve Sun long and lat at time t
3. check whether Earth is shadowing or not

Nonetheless, as there is a function which does exactly what I need, I wanted to be sure before implementing it.
Thanks in advance.

**Comment by dododo42** [comment, marked as answer]
I just realised that is_sunlit(ephemeris) refers to the ephemeris of the Sun.
The following code should return the right answer:

`from skyfield.api import EarthSatellite, load`
`ts = load.timescale()`
`satellite = EarthSatellite(line1, line2, name, ts)`
`t = ts.utc(utc_date)`
`eph = load('de421.bsp')`
`sunlit = satellite.at(t).is_sunlit(eph)`

Hoping I have not wasted anyone's time.

**Reply by brandon-rhodes** [reply]
Thank you for following up with the solution! Is there anywhere the documentation could have explained that more clearly, and made the purpose of the `ephemeris` argument clearer?

**Reply by dododo42** [reply]
As I come from more analytical environments, my mistake was not knowing that de421.bsp is a standard file for planets ephemiris.
I originally thought it was an example file of some object just to showcase how to use `is_sunlit()`.
Nevertheless, I think the documentation is good already as it is. Only some inexperience from my side.
